python-phoenixdb/phoenixdb/tests/test_sqlalchemy.py

```diff
@@ -52,6 +52,39 @@ def test_textual(self):
             finally:
                 connection.execute('drop table if exists ALCHEMY_TEST')
 
+    def test_schema_filtering(self):
+        engine = self._create_engine()
+        with engine.connect() as connection:
+            try:
+                connection.execute('drop table if exists ALCHEMY_TEST')
+                connection.execute('drop table if exists A.ALCHEMY_TEST_A')
+                connection.execute('drop table if exists B.ALCHEMY_TEST_B')
+
+                connection.execute(text('create table ALCHEMY_TEST (ID integer primary key)'))
+                connection.execute(text('create table A.ALCHEMY_TEST_A (ID_A integer primary key)'))
+                connection.execute(text('create table B.ALCHEMY_TEST_B (ID_B integer primary key)'))
+
+                inspector = db.inspect(engine)
+
+                self.assertEqual(inspector.get_schema_names(), [None, 'A', 'B', 'SYSTEM'])
```

Comment:
Is returning `None` vs `''` (empty string) intentional? i.e. is this convention for sqlalchemy?

Reply:
I am not actually sure. The documentation doesn't specify this.

The internal representation in Phoenix is null, which translates to None.
When specifying the schema filter, the default schema is also represented in sqlAlchemy by None.

Based on this, returning None here seems to make sense. 

I'll try to figure out what i.e. the mysql driver returns here, and adjust if it returns ''

Reply:
> The internal representation in Phoenix is null, which translates to None.

So, Phoenix is really weird when it comes to "null" and "empty string", in that they are functionally equivalent. Not sure if that's a surprise to you.

That all said, `None` is fine. Mostly curious :)

Reply:
In most DBs the default scheme seems to have textual representation, so I couldn't find anything definitive by a cursory search. 

At least we also return None from **sqlalchemy.engine.reflection.Inspector.default_schema_name** , so we are internally consistent.

I'll try to check how Hue handles this semantics tomorrow. (Should have done that before opening the PR)

I've run into the null = '' weirdness before, but it seems to work in direction only. (i.e. `column = ''` won't match on empty strings / null , only `is null` will)